src/sentry/static/sentry/app/views/settings/project/projectOwnership/selectOwners.tsx

```diff
@@ -15,88 +14,84 @@ import TeamStore from 'app/stores/teamStore';
 import IdBadge from 'app/components/idBadge';
 import MultiSelectControl from 'app/components/forms/multiSelectControl';
 import ActorAvatar from 'app/components/avatar/actorAvatar';
-import SentryTypes from 'app/sentryTypes';
 import Button from 'app/components/button';
 import {IconAdd} from 'app/icons';
 import Tooltip from 'app/components/tooltip';
-
-class ValueComponent extends React.Component {
-  static propTypes = {
-    value: PropTypes.object,
-    onRemove: PropTypes.func,
-  };
-
-  handleClick = () => {
-    this.props.onRemove(this.props.value);
-  };
-
-  render() {
-    return (
-      <a onClick={this.handleClick}>
-        <ActorAvatar actor={this.props.value.actor} size={28} />
-      </a>
-    );
-  }
+import withProjects from 'app/utils/withProjects';
+import withApi from 'app/utils/withApi';
+import {Actor, Member, Organization, Project, Team, User} from 'app/types';
+
+export type Owner = {
+  value: string;
+  label: React.ReactNode;
+  searchKey: string;
+  actor: Actor;
+  disabled?: boolean;
+};
+
+type ValueProps = {
+  value: Owner;
+  onRemove: (item: Owner) => void;
+};
+
+function ValueComponent({value, onRemove}: ValueProps) {
+  return (
+    <a onClick={() => onRemove(value)}>
+      <ActorAvatar actor={value.actor} size={28} />
+    </a>
+  );
 }
 
-const getSearchKeyForUser = user =>
+const getSearchKeyForUser = (user: User) =>
   `${user.email && user.email.toLowerCase()} ${user.name && user.name.toLowerCase()}`;
 
-export default class SelectOwners extends React.Component {
-  static propTypes = {
-    project: SentryTypes.Project,
-    organization: SentryTypes.Organization,
-    value: PropTypes.array,
-    onChange: PropTypes.func,
-    onInputChange: PropTypes.func,
-    disabled: PropTypes.bool,
-  };
-
-  constructor(...args) {
-    super(...args);
-    this.api = new Client();
-
-    // See comments in `handleAddTeamToProject` for why we close the menu this way
-    this.projectsStoreMixin = Reflux.listenTo(ProjectsStore, () => {
-      this.closeSelectMenu();
-    });
-  }
-
+type Props = {
+  api: Client;
+  organization: Organization;
+  project: Project;
+  projects: Project[];
+  value: any;
+  onChange: (owners: Owner[]) => void;
+  disabled: boolean;
+  onInputChange?: (text: string) => void;
+};
+
+type State = {
+  loading: boolean;
+  inputValue: string;
+};
+
+class SelectOwners extends React.Component<Props, State> {
   state = {
     loading: false,
     inputValue: '',
   };
 
-  componentDidMount() {
-    if (this.projectsStoreMixin) {
-      this.projectsStoreMixin.componentDidMount();
+  componentDidUpdate(prevProps: Props) {
+    // Once a team has been added to the project the menu can be closed.
+    if (!isEqual(this.props.projects, prevProps.projects)) {
+      this.closeSelectMenu();
     }
   }
 
-  componentWillUnmount() {
-    this.api = null;
+  private selectRef = React.createRef<React.Instance>();
```

Comment:
What's this Instance? 🤔

Reply:
For some bizarre reason the ref gets passed a component instance, which is why we need to use findDOMNode instead of just working with the HTMLElement directly.